### PR TITLE
AST-1425 - Added new color palette preset options

### PR DESCRIPTION
**Pull request by imnavanath**
### Description
- Added color palette preset options

### Screenshots
- https://share.bsf.io/NQu7nPBK

### Types of changes
- New feature 

### How has this been tested?
- Check with color-palette
- Adjust settings from color picker

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: inc/theme-update/astra-update-functions.php [hunk 1]
```diff
@@ -3274,3 +3274,20 @@ function astra_improve_gutenberg_editor_ui() {
...
+ */
+function astra_add_color_palette_presets() {
+	$astra_color_palette = get_option( 'astra-color-palettes', array() );
+
+	// Check if Presets array is already set or not. If not then set it as array.
+	if ( ! isset( $astra_color_palette['presets'] ) && ! is_array( $astra_color_palette['presets'] ) ) {
+		$astra_color_palette['presets'] = astra_get_palette_presets();
+		update_option( 'astra-color-palettes', $astra_color_palette );
+	}
+}
```
Comment (premanshup) [inline inc/theme-update/astra-update-functions.php:3293]: @imnavanath I think we should create new file for this. Line of code here is a lot.

Comment (imnavanath) [reply inc/theme-update/astra-update-functions.php:3293]: Yes I agree.

Comment (premanshup) [reply inc/theme-update/astra-update-functions.php:3293]: @imnavanath Can you please create a new file and add these there. Let's add these functions in the file for future updates.

File: inc/extras.php [hunk 2]
```diff
@@ -801,6 +801,72 @@ function astra_check_elementor_pro_3_5_version() {
...
+			'#2B6CB0',
+			'#215387',
+			'#222222',
+			'#3B3B3B',
+			'#515151',
+			'#626262',
+			'#E1E1E1',
+			'#F7F7F7',
+			'#FFFFFF',
+		),
```
Comment (premanshup) [inline inc/extras.php:822]: @imnavanath Please confirm all the color combinations we should add here from design team.

File: inc/customizer/extend-custom-controls/src/color-palette/color-palette.js [hunk 3]
```diff
@@ -153,6 +163,57 @@ const ColorPaletteComponent = (props) => {
...
+
+		if( updateState.isVisible = true ) {
+			updateState.isVisible = false;
+			updateValues(updateState);
+		}
+	};
+
+	var presetOptions = (
+		<>
+			<Popover position="bottom center" className="" onClose={toggleClose}>
```
Comment (premanshup) [inline inc/customizer/extend-custom-controls/src/color-palette/color-palette.js:188]:
```suggestion
			<Popover position="bottom center" onClose={toggleClose}>
```

File: inc/customizer/extend-custom-controls/src/color-palette/color-palette.js [hunk 4]
```diff
@@ -153,6 +163,57 @@ const ColorPaletteComponent = (props) => {
...
+				<div className="">
```
Comment (premanshup) [inline inc/customizer/extend-custom-controls/src/color-palette/color-palette.js:189]:
```suggestion
```
@imnavanath Please check if this empty div can be removed?

File: inc/customizer/extend-custom-controls/src/color-palette/color-palette.js [hunk 5]
```diff
@@ -88,10 +88,20 @@ const ColorPaletteComponent = (props) => {
 		updateValues(updateState);
 	};
 
+	const handlePresetAssignment = (presetKey) => {
+		if ( state.presets && state.presets[presetKey] ) {
+			state.presets[presetKey].map( ( item, index ) => {
+				if ( item ) {
```
Comment (premanshup) [inline inc/customizer/extend-custom-controls/src/color-palette/color-palette.js:94]:
```suggestion
```
@imnavanath I think ideally `item` should always contain the value. Please check if we can remove this?

File: inc/customizer/extend-custom-controls/src/color-palette/color-palette.js [hunk 6]
```diff
@@ -153,6 +163,57 @@ const ColorPaletteComponent = (props) => {
...
+
+	const toggleClose = () => {
+		let updateState = {
+			...state,
+		};
+
+		if( updateState.isVisible = true ) {
+			updateState.isVisible = false;
+			updateValues(updateState);
+		}
```
Comment (premanshup) [inline inc/customizer/extend-custom-controls/src/color-palette/color-palette.js:183]: @imnavanath Same as above.

File: inc/extras.php (hunk 2 above)
Comment (imnavanath) [reply inc/extras.php:822]: These are the dummy ones I added for first view only, after design review we can manipulate that.
Accordingly I've added "needs inputs" requirement in respective card & assigned that to Satishk.

File: inc/customizer/extend-custom-controls/src/color-palette/color-palette.js [hunk 7]
```diff
@@ -153,6 +163,57 @@ const ColorPaletteComponent = (props) => {
 		</>
 	);
 
+	const toggleVisible = () => {
+		let updateState = {
+			...state,
+		};
+
+		updateState.isVisible = true;
+		updateValues(updateState);
```
[the remark this answers is not in the document]
Comment (imnavanath) [reply inc/customizer/extend-custom-controls/src/color-palette/color-palette.js:172]: No actually, its member variable only, so its creating console error - https://share.bsf.io/X6u8dJr8 
So keeping it as it is.

File: inc/customizer/extend-custom-controls/src/color-palette/color-palette.js (hunk 6 above)
Comment (imnavanath) [reply inc/customizer/extend-custom-controls/src/color-palette/color-palette.js:183]: No actually, its member variable only, so its creating console error - https://share.bsf.io/X6u8dJr8
So keeping it as it is.

File: inc/customizer/extend-custom-controls/src/color-palette/color-palette.js (hunk 3 above)
Comment (imnavanath) [reply inc/customizer/extend-custom-controls/src/color-palette/color-palette.js]: Done

File: inc/customizer/extend-custom-controls/src/color-palette/color-palette.js (hunk 4 above)
Comment (imnavanath) [reply inc/customizer/extend-custom-controls/src/color-palette/color-palette.js]: Done

File: inc/customizer/extend-custom-controls/src/color-palette/color-palette.js (hunk 5 above)
Comment (imnavanath) [reply inc/customizer/extend-custom-controls/src/color-palette/color-palette.js]: Yes removed 👍

File: inc/extras.php (hunk 2 above)
Comment (imnavanath) [reply inc/extras.php:822]: Already asked designers to provide colors so later we will just replace it - https://wp-astra.atlassian.net/browse/AST-1335?focusedCommentId=17704

File: inc/theme-update/astra-update-functions.php (hunk 1 above)
Comment (imnavanath) [reply inc/theme-update/astra-update-functions.php:855]: Done 👍 https://github.com/brainstormforce/astra/pull/3840/commits/0edde618890213df5e924bdd2970f3996efb9e54

**Pull request merged**